pymc3/distributions/continuous.py

```diff
@@ -434,6 +434,202 @@ def _repr_latex_(self, name=None, dist=None):
                                                                 get_variable_name(mu),
                                                                 get_variable_name(sd))
 
+class TruncatedNormal(Continuous):
+    R"""
+    Univariate truncated normal log-likelihood.
+
+    The pdf of this distribution is
+
+    .. math::
+
+       f(x;\mu ,\sigma ,a,b)={\frac {\phi ({\frac {x-\mu }{\sigma }})}{
+       \sigma \left(\Phi ({\frac {b-\mu }{\sigma }})-\Phi ({\frac {a-\mu }{\sigma }})\right)}}
+
+    Truncated normal distribution can be parameterized either in terms of precision
+    or standard deviation. The link between the two parametrizations is
+    given by
+
+    .. math::
+
+       \tau = \dfrac{1}{\sigma^2}
+
+
+    .. plot::
+
+        import matplotlib.pyplot as plt
+        import numpy as np
+        import scipy.stats as st
+        plt.style.use('seaborn-darkgrid')
+        x = np.linspace(-10, 10, 1000)
+        mus = [0.,  0., 0.]
+        sds = [3.,5.,7.]
+        a1 = [-3, -5, -5]
+        b1 = [7, 5, 4]
+        for mu, sd, a, b in zip(mus, sds,a1,b1):
+            print mu, sd, a, b
+            an, bn = (a - mu) / sd, (b - mu) / sd
+            pdf = st.truncnorm.pdf(x, an,bn, loc=mu, scale=sd)
+            plt.plot(x, pdf, label=r'$\mu$ = {}, $\sigma$ = {}, a={}, b={}'.format(mu, sd, a, b))
+        plt.xlabel('x', fontsize=12)
+        plt.ylabel('f(x)', fontsize=12)
+        plt.legend(loc=1)
+        plt.show()
+
+    ========  ==========================================
+    Support   :math:`x \in [a, b]`
+    Mean      :math:`\mu +{\frac {\phi (\alpha )-\phi (\beta )}{Z}}\sigma`
+    Variance  :math:`\sigma ^{2}\left[1+{\frac {\alpha \phi (\alpha )-\beta \phi (\beta )}{Z}}-
+    \left({\frac {\phi (\alpha )-\phi (\beta )}{Z}}\right)^{2}\right]`
+    ========  ==========================================
+
+    Parameters
+    ----------
+    mu : float
+        Mean.
+    sd : float
+        Standard deviation (sd > 0).
+    a : float (optional)
+        Left bound.
+    b : float (optional)
+        Right bound.
+
+    Examples
+    --------
+    .. code-block:: python
+
+        with pm.Model():
+            x = pm.TruncatedNormal('x', mu=0, sd=10, a=0)
+
+        with pm.Model():
+            x = pm.TruncatedNormal('x', mu=0, sd=10, b=1)
+
+        with pm.Model():
+            x = pm.TruncatedNormal('x', mu=0, sd=10, a=0, b=1)
+
+    """
+
+    def __init__(self, mu=0, sd=None, tau=None, a=None, b=None, **kwargs):
+        tau, sd = get_tau_sd(tau=tau, sd=sd)
+        self.sd = tt.as_tensor_variable(sd)
+        self.tau = tt.as_tensor_variable(tau)
+        self.a = tt.as_tensor_variable(a) if a is not None else a
+        self.b = tt.as_tensor_variable(b) if b is not None else b
+        self.mu =  tt.as_tensor_variable(mu)
+        self.mean = self.mu
```

Comment:
this isn't the mean of a truncated normal, and may cause problems when it is used as an initial point

Reply:
Sure, no problem, wasn't sure where it is used. Do we need also proper implementations of median and mode?

Reply:
looks good! no need to implement the others (unless it is trivial).  it isn't documented too well, but each distribution has `defaults` defined as a list, and iterates through to find one that has finite log probability to initialize with.  I think the "default" for `defaults` is `[mean, median, mode]`.